ospfd/ospf_vty.c

```diff
@@ -72,13 +72,13 @@ int str2area_id(const char *str, struct in_addr *area_id, int *area_id_fmt)
 	return 0;
 }
 
-void area_id2str(char *buf, int length, struct in_addr *area_id,
-		 int area_id_fmt)
+static void area_id2str(char *buf, int length, struct in_addr *area_id,
+			int area_id_fmt)
 {
 	memset(buf, 0, length);
 
 	if (area_id_fmt == OSPF_AREA_ID_FMT_DOTTEDQUAD)
-		strncpy(buf, inet_ntoa(*area_id), length);
+		strncpy(buf, inet_ntoa(*area_id), length - 1);
```

Comment:
Is there any special requirement to use `strncpy` here? Like expecting to have the buffer tail zeroed out entirely?

Otherwise I'd suggest to remove the `memset` and use `strlcpy` or `snprintf` here. `strncpy` is usually ineficient and error prone.